src/routes/RequestsRoute.js

```diff
@@ -1226,8 +1269,13 @@ class RequestsRoute extends React.Component {
   );
 
   onBeforePrintForSinglePrintButton = (requestId) => {
-    this.savePrintEventDetails([requestId]); // TODO: add enable print log settings check
-  }
+    const isViewPrintDetailsEnabled =
+      get(this.props.resources, viewPrintDetailsPath) === 'true';
+
+    if (isViewPrintDetailsEnabled) {
+      this.savePrintEventDetails([requestId]);
+    }
+  };
```

Comment:
Please check if this validation needs to be moved to getListFormatter SinglePrint,, similar to onBeforePrintForPrintButton ?